### PR TITLE
feat: Add owner dropdown to search prices

### PR DESCRIPTION
**Pull request by ShaBloze**
## Description

This PR is about being able to search by an owner, by getting all market participants from the charges domain.

Also added some additional features:
- Add sort on all columns
- Clear button should also clear the table result.

## References

- #1656
https://app.zenhub.com/workspaces/team-volt-60a61059028fd7000e49956e/issues/energinet-datahub/geh-charges/1656

### CODE REVIEW

File: libs/dh/charges/feature-prices/src/lib/search-result/dh-charges-prices-result.component.html [hunk 1]
```diff
@@ -46,28 +46,32 @@ <h3>
     <table
       aria-label="prices search results"
       mat-table
-      [dataSource]="this.dataSource"
+      [dataSource]="dataSource"
       matSort
+      matSortActive="chargeName"
+      matSortDirection="asc"
     >
       <!-- price id column -->
```
Comment (HenrikSommer) [inline libs/dh/charges/feature-prices/src/lib/search-result/dh-charges-prices-result.component.html:54]: Let's update the comment or remove it

File: libs/dh/charges/feature-prices/src/lib/search-result/dh-charges-prices-result.component.html [hunk 2]
```diff
@@ -46,28 +46,32 @@ <h3>
...
-        <th mat-header-cell *matHeaderCellDef>
-          {{ t("priceId") }}
+      <ng-container matColumnDef="chargeId">
+        <th *matHeaderCellDef mat-header-cell mat-sort-header>
+          {{ t("chargeId") }}
         </th>
         <td mat-cell *matCellDef="let element">{{ element.chargeId }}</td>
       </ng-container>
 
       <!-- price name column -->
```
Comment (HenrikSommer) [inline libs/dh/charges/feature-prices/src/lib/search-result/dh-charges-prices-result.component.html:62]: Let's update the comment or remove it

File: libs/dh/charges/feature-prices/src/lib/search-result/dh-charges-prices-result.component.html [hunk 3]
```diff
@@ -99,34 +103,38 @@ <h3>
```
Comment (HenrikSommer) [inline libs/dh/charges/feature-prices/src/lib/search-result/dh-charges-prices-result.component.html:103]: No sorting by `taxIndicator` or `transparentInvoicing`? - I'm not sure it is needed, just asking :-)

File: libs/dh/shared/domain/src/lib/generated/v1/model/market-participant-v1-dto.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+ * 
+ *
+ * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
+ * https://openapi-generator.tech
+ * Do not edit the class manually.
+ */
+
+
+export interface MarketParticipantV1Dto { 
+    id: string;
```
Comment (HenrikSommer) [inline libs/dh/shared/domain/src/lib/generated/v1/model/market-participant-v1-dto.ts:15]: Should this not be an actual guid?

File: libs/dh/charges/feature-prices/src/lib/search-result/dh-charges-prices-result.component.html (hunk 3 above)
Comment (ShaBloze) [reply libs/dh/charges/feature-prices/src/lib/search-result/dh-charges-prices-result.component.html:103]: For now it is not needed :)

**Review by HenrikSommer — APPROVED** [review]

**Pull request merged**